benchmarks/hough_transform/oop_c#/Program.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Drawing;
+using System.Linq;
+
+namespace oop_c_
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var hough = new HoughTransform("benchmarks/hough_transform/Pentagon.png", 480, 640);
+            var output = hough.ComputeTransformation();
+            System.Console.WriteLine(output.Cast<int>().Sum());
+        }
+    }
+
+    public class HoughTransform
+    {
+
+        int[,] transformed { get; set; }
+        PictureData pictureData { get; set; }
+        int rhoAxisSize { get; }
+        int thetaAxisSize { get; }
+        public HoughTransform(string filename, int rhoAxisSize, int thetaAxisSize)
+        {
+            pictureData = new PictureData(filename);
+            this.rhoAxisSize = rhoAxisSize;
+            this.thetaAxisSize = thetaAxisSize;
+            transformed = new int[thetaAxisSize, rhoAxisSize];
+        }
+
+        public int[,] ComputeTransformation()
+        {
+            int width = pictureData.Width, height = pictureData.Height;
+            int diagonal = (int)(Math.Ceiling(MyMath.Hypotenuse(width, height)));
+            int halfRhoAxisSize = rhoAxisSize / 2;
+            (double[] sinTable, double[] cosTable) = MyMath.CosSinRadianTables(thetaAxisSize);
+
+
+            // Scanning through each (x,y) pixel of the image
+            for (int y = 0; y < height; y++)
+            {
+                for (int x = 0; x < width; x++)
+                {
+                    Color color = pictureData.GetPixelColor(x, y);
+                    
+                    // If a pixel is white, skip it.
+                    if (color.Name == "ffffffff")
+                        continue;
+
+                    // If it is black (an edge pixel), loop through all possible values of θ, 
+                    // calculate the corresponding ρ, find the θ and ρ index in the 
+                    // accumulator, and increment the accumulator base on those index pairs.
+                    for (int theta = 0; theta < thetaAxisSize; theta++)
+                    {
+                        // Distance from the origin to the closest point on the straight line

```

Comment:
Hvad er det for en lige linje som vi finder det tætteste punkt på?

Reply:
Hver θ (theta) og ρ (rho) værdierne repræsenterer sammen med (x,y) koordinaterne en linje. Rho er så afstanden mellem origo (0,0) og det sted på linjen der er tættest på origo. Så hver theta værdi er en ny linje vi plotter ind i vores resultat.